vmhost/contexts/managedType.go

```diff
@@ -657,3 +668,110 @@ func (context *managedTypesContext) WriteManagedVecOfManagedBuffers(
 	metering := context.host.Metering()
 	metering.UseAndTraceGas(sumOfItemByteLengths * metering.GasSchedule().BaseOperationCost.DataCopyPerByte)
 }
+
+// NewManagedMap creates a new empty managed map in the managed buffers map and returns the handle
+func (context *managedTypesContext) NewManagedMap() int32 {
+	newHandle := int32(len(context.managedTypesValues.mMapValues))
+	for {
+		if _, ok := context.managedTypesValues.mMapValues[newHandle]; !ok {
+			break
+		}
+		newHandle++
+	}
+	newmMap := make(map[string][]byte, 0)
+	context.managedTypesValues.mMapValues[newHandle] = newmMap
+	return newHandle
+}
+
+// ManagedMapPut puts the key and value bytes stored at those respective handles in the map
+func (context *managedTypesContext) ManagedMapPut(mMapHandle int32, keyHandle int32, valueHandle int32) error {
+	mMap, ok := context.managedTypesValues.mMapValues[mMapHandle]
+	if !ok {
+		return vmhost.ErrNoManagedMapUnderThisHandle
+	}
+
+	key, err := context.GetBytes(keyHandle)
+	if err != nil {
+		return err
+	}
+	keyCopy := make([]byte, len(key))
+	copy(keyCopy, key)
+
+	value, err := context.GetBytes(valueHandle)
+	if err != nil {
+		return err
+	}
+	valueCopy := make([]byte, len(value))
+	copy(valueCopy, value)
+
+	mMap[string(keyCopy)] = valueCopy
+
+	return nil
+}
+
+// ManagedMapGet gets the bytes stored as the key handle in an output value handle
+func (context *managedTypesContext) ManagedMapGet(mMapHandle int32, keyHandle int32, outValueHandle int32) error {
+	_, _, value, _, err := context.getKeyValueFromManagedMap(mMapHandle, keyHandle)
+	if err != nil {
+		return err
+	}
+
+	err = context.setBytesIfBufferCreated(value, outValueHandle)
```

Comment:
Also I think just SetBytes is enough, it will create the buffer if it was not created, which is ok.
This eliminates the need to also call new buffer, which is a performance improvement

Reply:
done